daemon/containerd/image_list.go

```diff
@@ -66,63 +70,32 @@ func (i *ImageService) Images(ctx context.Context, opts types.ImageListOptions)
 		root = make([]*[]digest.Digest, len(imgs))
 		layers = make(map[digest.Digest]int)
 	}
+
 	for n, img := range imgs {
 		if !filterFn(img) {
 			continue
 		}
 
-		diffIDs, err := img.RootFS(ctx)
+		platforms, err := images.Platforms(ctx, i.client.ContentStore(), img.Target)
 		if err != nil {
 			return nil, err
 		}
-		chainIDs := identity.ChainIDs(diffIDs)
-		if opts.SharedSize {
-			root[n] = &chainIDs
-			for _, id := range chainIDs {
-				layers[id] = layers[id] + 1
-			}
-		}
 
-		size, err := img.Size(ctx)
-		if err != nil {
-			return nil, err
-		}
+		for m, platform := range platforms {
+			image, chainIDs, err := i.singlePlatformImage(ctx, img, platform, opts)
+			if err != nil {
+				return nil, err
+			}
 
-		virtualSize, err := computeVirtualSize(chainIDs, snapshotSizeFn)
-		if err != nil {
-			return nil, err
-		}
+			summaries = append(summaries, image)
 
-		ref, err := reference.ParseNormalizedNamed(img.Name())
-		if err != nil {
-			return nil, err
-		}
-
-		var repoDigests, repoTags []string
-		if isDanglingImage(img) {
-			repoTags = []string{"<none>:<none>"}
-			repoDigests = []string{"<none>@<none>"}
-		} else {
-			familiarName := reference.FamiliarString(ref)
-			repoTags = []string{familiarName}
-			repoDigests = []string{familiarName + "@" + img.Target().Digest.String()} // "hello-world@sha256:bfea6278a0a267fad2634554f4f0c6f31981eea41c553fdf5a83e95a41d40c38"
+			if opts.SharedSize {
+				root[n+m] = &chainIDs
```

Comment:
(follow-up to 1 comment)
```suggestion
				root = append(root, &chainIDs)
```